app/jobs/inat_import_job.rb

```diff
@@ -0,0 +1,408 @@
+    add_provisional_naming # iNat provisionals are not identifications
+    adjust_consensus_name_naming # also adds naming for provisionals
+
+    Observation::NamingConsensus.new(@observation).calc_consensus
+  end
+
+  def add_identifications_with_namings
+    @inat_obs.inat_identifications.each do |identification|
+      inat_taxon = ::Inat::Taxon.new(identification[:taxon])
+      next if name_already_proposed?(inat_taxon.name)
+
+      add_naming_with_vote(name: inat_taxon.name)
+    end
+  end
+
+  def name_already_proposed?(name)
+    Naming.where(observation_id: @observation.id).
+      map(&:name).include?(name)
+  end
+
+  def add_naming_with_vote(name:, user: inat_manager, value: Vote::MAXIMUM_VOTE)
+    naming = Naming.create(observation: @observation,
+                           user: user, name: name)
+
+    Vote.create(naming: naming, observation: @observation,
+                user: user, value: value)
+  end
+
+  def add_provisional_naming
+    nom_prov = @inat_obs.provisional_name
+    return if nom_prov.blank?
+
+    # NOTE: There will be >= 1 match because of add_missing_provisional_name.
+    # If a provisional Name was added during import, use it;
+    # (It's the most recently created, and won't be deprecated.)
+    # else grab another matching one.
+    name = best_mo_homonym(nom_prov)
+    add_naming_with_vote(name: name)
+  end
+
+  def best_mo_homonym(text_name)
+    Name.where(text_name: text_name).
+      order(deprecated: :asc, created_at: :desc).
+      first
+  end
+
+  def adjust_consensus_name_naming
+    naming = Naming.find_by(observation: @observation,
+                            name: @observation.name)
+
+    if naming.nil?
+      add_naming_with_vote(name: @observation.name,
+                           user: inat_manager, value: Vote::MAXIMUM_VOTE)
+    else
+      vote = Vote.find_by(naming: naming, observation: @observation)
+      vote.update(value: Vote::MAXIMUM_VOTE)
+    end
+  end
+
+  def lat_lon_accuracy
+    "#{@inat_obs.inat_location} " \
+    "+/-#{@inat_obs.inat_public_positional_accuracy} m"
+  end
+
+  def add_inat_sequences
+    @inat_obs.sequences.each do |sequence|
+      params = { action: :sequence, method: :post,
+                 api_key: inat_manager_key.key,
+                 observation: @observation.id,
+                 locus: sequence[:locus],
+                 bases: sequence[:bases],
+                 archive: sequence[:archive],
+                 accession: sequence[:accession],
+                 notes: sequence[:notes] }
+
+      # NOTE: Error handling? 2024-06-19 jdc.
+      # https://github.com/MushroomObserver/mushroom-observer/issues/2382
+      API2.execute(params)
+    end
+  end
+
+  def add_snapshot_of_import_comment
+    params = { target: @observation, user: inat_manager,
+               summary: "#{:inat_data_comment.t} #{@observation.created_at}",
+               comment: import_snapshot }
+    Comment.create(params)
+  end
+
+  def import_snapshot
```

Comment:
For this ABC metric violationm I'd move most of this into the `InatObs` object as a `comment` method or some such.  That would probably go a long way towards addressing this since you'd no longer be doing all those dereferences.